src/main/java/cubicchunks/asm/mixin/core/client/MixinRenderGlobal_BiggerRenderChunk_Common.java

```diff
@@ -0,0 +1,230 @@
+                    ClassInheritanceMultiMap<Entity>[] entityLists = world.getChunkFromChunkCoords(cx, cz).getEntityLists();
+                    for (int cy = cy0; cy < cy0 + renderChunkCubeSize && cy < 16 && cy >= 0; cy++) {
+                        mutableEntityMapWrapper.values.addAll(entityLists[cy].values);
+                    }
+                }
+        }
+        return BlockPos.ORIGIN;
+    }
+
+    @Redirect(method = "renderEntities", at = @At(value = "INVOKE", target = "Lnet/minecraft/world/chunk/Chunk;getEntityLists()[Lnet/minecraft/util/ClassInheritanceMultiMap;"))
+    public ClassInheritanceMultiMap<Entity>[] onGettingClassInheritanceMultiMap(Chunk chunk) {
+        dummyEntityStorageArray[0] = mutableEntityMapWrapper;
+        return dummyEntityStorageArray;
+    }
+
+    /**
+     * Overwrite getRenderChunk(For)Offset to support extended height.
+     *
+     * @author Barteks2x
+     * @reason Remove hardcoded height checks, it's a simple method and doing it differently would be problematic and confusing (Inject with local capture into BlockPos.getX() and redirect of BlockPos.getY())
+     */
+    @Nullable
+    @Overwrite
+    private RenderChunk getRenderChunkOffset(BlockPos playerPos, RenderChunk renderChunkBase, EnumFacing facing) {
+        BlockPos blockpos = renderChunkBase.getBlockPosOffset16(facing);
+        int size = RenderVariables.getRenderChunkSize();
+        return MathHelper.abs(playerPos.getX() - blockpos.getX()) > this.renderDistanceChunks * size ? null
+                : MathHelper.abs(playerPos.getY() - blockpos.getY()) > this.renderDistanceChunks * size ? null
+                        : MathHelper.abs(playerPos.getZ() - blockpos.getZ()) > this.renderDistanceChunks * size ? null
+                                : this.viewFrustum.getRenderChunk(blockpos);
+    }
+
+    @ModifyConstant(method = "renderWorldBorder", constant = {
+            @Constant(doubleValue = 0.0D),
+            @Constant(doubleValue = 256.0D)
+    }, slice = @Slice(from = @At(value = "HEAD"), to = @At(value = "INVOKE", target = "Lnet/minecraft/client/renderer/Tessellator;draw()V")), require = 2)
+    private double renderWorldBorder_getRenderHeight(double original, Entity entity, float partialTicks) {
+        return original == 0.0D ? entity.posY - 128 : entity.posY + 128;
+    }
+
+    @Redirect(method = "setupTerrain", at = @At(value = "INVOKE", target = "Lnet/minecraft/client/renderer/chunk/RenderChunk;needsUpdate()Z"))
+    public boolean doChunkNeedUpdate(RenderChunk renderChunk) {
+        int maxPos = RenderVariables.getRenderChunkMaxPos();
+        return renderChunk.needsUpdate() && world.isAreaLoaded(renderChunk.getPosition(),
+                renderChunk.getPosition().add(maxPos, maxPos, maxPos));
+    }
+
+    @ModifyConstant(method = "setupTerrain", constant = @Constant(doubleValue = 16.0D))
+    public double onSetupTerrain1(double oldValue) {
+        return Double.MAX_VALUE;
```

Comment:
What does that do?

Reply:
Removing unnecessary check `d0 * d0 + d1 * d1 + d2 * d2 > 16.0D`